chromadb/api/models/CollectionCommon.py

```diff
@@ -508,39 +484,63 @@ def _validate_and_prepare_upsert_request(
         documents: Optional[OneOrMany[Document]],
         images: Optional[OneOrMany[Image]],
         uris: Optional[OneOrMany[URI]],
-    ) -> Tuple[
-        IDs,
-        Embeddings,
-        Optional[Metadatas],
-        Optional[Documents],
-        Optional[URIs],
-    ]:
-        (
-            ids,
-            embeddings,
-            metadatas,
-            documents,
-            images,
-            uris,
-        ) = self._validate_embedding_set(
-            ids, embeddings, metadatas, documents, images, uris
+        require_embeddings_or_data: bool = True,
+    ) -> RecordSet:
+        unpacked_embedding_set = self._unpack_record_set(
+            ids=ids,
+            embeddings=embeddings,
+            metadatas=metadatas,
+            documents=documents,
+            images=images,
+            uris=uris,
         )
 
-        if embeddings is None:
-            if documents is not None:
-                embeddings = self._embed(input=documents)
-            else:
-                embeddings = self._embed(input=images)
+        normalized_embeddings = (
+            self._normalize_embeddings(unpacked_embedding_set["embeddings"])
+            if unpacked_embedding_set["embeddings"] is not None
+            else None
+        )
 
-        return ids, embeddings, metadatas, documents, uris
+        self._validate_embedding_set(
+            ids=unpacked_embedding_set["ids"],
+            embeddings=normalized_embeddings,
+            metadatas=unpacked_embedding_set["metadatas"],
+            documents=unpacked_embedding_set["documents"],
+            images=unpacked_embedding_set["images"],
+            uris=unpacked_embedding_set["uris"],
+            require_embeddings_or_data=require_embeddings_or_data,
+        )
+
+        prepared_embeddings = None
+        try:
+            prepared_embeddings = self._compute_embeddings(
+                embeddings=normalized_embeddings,
+                documents=unpacked_embedding_set["documents"],
+                images=unpacked_embedding_set["images"],
+                uris=None,
+            )
+        except ValueError:
```

Comment:
We should never swallow an exception! This could get raised if both `documents` and `images` are `None`. 

Catching it here is smart, because it lets us inform the user about what's really going on - let's instead raise an exception with a different message that tells the user they need to give us one of `images` or `documents`.

Reply:
This now catches all `ValueError` and only raises if `require_embeddings_or_data` is `True`. What happens when both `documents` and `images` are `None`, but `require_embeddings_or_data` is `False` , for example on a call to `Collection.update` ? 

`_compute_embeddings` will raise `ValueError`, which will then get caught here and swallowed. Then `prepared_embeddings` will be whatever `normalized_embeddings` is, which may be `None` since `embeddings` is `Optional`. This is accidental and we should be rigorous / intentional with what we're passing around and when, and never swallow exceptions.

It's ok for all of `embeddings`, `documents`, `images`, and `uris` to be `None` on a call to `Collection.update` since we may only be updating the `metadatas`. I think the right thing is to not call `_compute_embeddings` if this is the case and `require_embeddings_or_data` is `False`.